### ISSUE TITLE
Add support for IPv6 zone identifier in nameserver ip address (resolv.conf)

### ISSUE THREAD

**Issue by csarn** [NLnetLabs/ldns#176]
I have a network where I get an IPv6 nameserver via router advertisement, and it is a link-local ip. So the resolv.conf entry looks something like this:

```
nameserver fe80::1%eth0
```

Now ssh uses ldns to look up SSHFP records when having `VerifyHostKeyDNS yes`. It tries the lookup 3 times, each one timing out, then ssh fails:

```
$ strace -e sendto,poll ssh -vvv some.host.name
(...)
debug3: verify_host_key_dns
sendto(4, "<redacted>"..., 41, 0, {sa_family=AF_INET6, sin6_port=htons(53), sin6_flowinfo=htonl(0), inet_pton(AF_INET6, "fe80::1", &sin6_addr), sin6_scope_id=0}, 28) = 41
poll([{fd=4, events=POLLIN|POLLERR}], 1, 5000) = 0 (Timeout)
sendto(4, "<redacted>"..., 41, 0, {sa_family=AF_INET6, sin6_port=htons(53), sin6_flowinfo=htonl(0), inet_pton(AF_INET6, "fe80::1", &sin6_addr), sin6_scope_id=0}, 28) = 41
poll([{fd=4, events=POLLIN|POLLERR}], 1, 5000) = 0 (Timeout)
sendto(4, "<redacted>"..., 41, 0, {sa_family=AF_INET6, sin6_port=htons(53), sin6_flowinfo=htonl(0), inet_pton(AF_INET6, "fe80::1", &sin6_addr), sin6_scope_id=0}, 28) = 41
poll([{fd=4, events=POLLIN|POLLERR}], 1, 5000) = 0 (Timeout)
DNS lookup error: general failure
(...)
Host key verification failed.
```

I assume that the relevant piece in the strace output is the `sin6_scope_id=0`, which is not valid.

I am running ldns 1.8.1-1 (from arch linux), openssh 9.0p1, on linux 5.17.8.

**Expected Behaviour:**
ldns sets the correct scope_id from the zone identifier in resolv.conf, resulting in successful DNS lookups from ssh.

**Comment by wtoorop** [comment]
I agree this should be done, but this is not easily accomplished with how things are currently handled. I guess we would need to add an additional list of IPv6 scope_id's to `struct ldns_struct_resolver`. I'll label this enhancement.
So not for the next release but the release thereafter...
@TCY16 would this be something you'd be willing to do?

**Comment by Crest** [comment]
This problem has been reported twice on the FreeBSD bug tracker already [PR#255316](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=255316), [PR#255316](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=255316). It's a real world problem, because people put link-local addresses in router advertisements. With a possible patch.

**Comment by patrickSpaceSurfer** [comment]
This should probably not be labelled an "enhancement", link local addresses are valid addresses and are often used as DNS servers (and routers) in IPv6 networks
